### DISCUSSION TITLE
question: how can i get encoded data with using useWriteContract

### DISCUSSION THREAD

**Discussion by yulafezmesi** [unicape/use-wagmi#142] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

1.1.0

### Current Behavior

Hey, thank you for great wrapper of wagmi hooks! 

I have a question about usage about `useWriteContract`. currently usage of quite usefull, however i want to use tenderly to simulation mode, so i need to get encoded data somehow, is there any way to achieve with current API's?

Thanks again!

**Comment by yulafezmesi** [comment]
actually, instead of encoding data, better to change config inside wagmi client to change RPC url to tenderly one. how can i do this on runtime?

**Comment by unicape** [comment]
Sorry, I didn't understand your question. Can you provide some use cases? I think it's easier to understand based on the code.

**Comment by unicape** [comment]
> actually, instead of encoding data, better to change config inside wagmi client to change RPC url to tenderly one. how can i do this on runtime?

As for this, you can modify it when creating wagmi

**Comment by yulafezmesi** [comment]
hey @unicape , thanks for the reply. dynamically changing only the RPC URL is not resolving for me because i needed to stay user's old connection (metamask) and sent all write/read operations through the tenderly connector ([mock](https://wagmi.sh/react/api/connectors/mock)) while simulation mode one. if the only RPC URL changing, the write operations still going to metamask.

if I conditionally set while initialize, the user can lost connection I believe. I built this but still need better approach. what do you thinking?

![image](https://github.com/unicape/use-wagmi/assets/45617686/99ab7194-a390-431c-acd6-005e500af86d)

**Comment by yulafezmesi** [comment]
on `ethers`, we could update this wallet client like this, basically I need to apply same thing on `wagmi`.

https://docs.tenderly.co/simulations-and-forks/integration-guides/instant-staging-qa-environment-for-dapps#create-a-tenderly-fork-via-the-ui

**Comment by unicape** [comment]
Do you want to use the Metamask connector when users connect and use other connectors when writing/reading?

**Comment by yulafezmesi** [comment]
actually i want to use another `config` in case simulation mode turned on but don't want to lose user's existing connection for eg: metamask. if i could change `connector` and `RPC url` in runtime, i think that would be solve the issue.

**Comment by unicape** [comment]
I want to know why not customize RPC in plugin

**Comment by unicape** [comment]
something like this

```js
const config = createConfig({
  chains: [mainnet],
  connectors: [
    metamask()
  ],
  transports: {
    [mainnet.id]: http(
      'https://rpc.tenderly.co/fork/YOUR_APIKEY',
    ),
  },
})

Vue.use(UseWagmiPlugin, { config })
```

**Comment by yulafezmesi** [comment]
when i do this, write operations still going metamask client and still asking for a signature :)

**Comment by unicape** [comment]
Not sure if this is what you want https://wagmi.sh/react/api/transports/unstable_connector

**Reply by yulafezmesi** [reply]
with `unstable_connector`, i can't provide the RPC url. so i believe it will use default RPC url.

**Comment by unicape** [comment]
I saw your question in the wagmi community. If you don't get a timely reply, you can try https://discord.com/invite/9zHPXuBpqy

**Reply by yulafezmesi** [reply]
thanks, i used discord as well however didn't got any response from them...

**Reply by unicape** [reply]
The wevm team is busy. . If the above method is most useful to you, you can use it first

**Reply by yulafezmesi** [reply]
yeap i understand that, no worries. i'll get things working either way, once i get proper solution, i'll implement.

also, how can i provide new config this? new `inject(WagmiConfigInjectionKey)` should be work but no luck. any idea?

https://github.com/unicape/use-wagmi/blob/main/packages/vue/src/hooks/useConfig.ts#L21
